File: 2730-find-the-longest-semi-repetitive-substring/2730-find-the-longest-semi-repetitive-substring.cpp
class Solution {
public:
    int longestSemiRepetitiveSubstring(string s) {
        int n=s.length();
        int cnt = 0;
        int mx=1;
        for(int i=0;i<n;i++){
            cnt=0;
            for(int j=i+1;j<n;j++){
                if(s[j-1] == s[j]){
                    cnt++;
                }
                if(cnt<=1){
                    mx=max(mx,j-i+1);
                }
            }
        }
        return mx;
        
    }
};

File: 2730-find-the-longest-semi-repetitive-substring/2730-find-the-longest-semi-repetitive-substring.js
/**
 * @param {string} s
 * @return {number}
 */
var longestSemiRepetitiveSubstring = function(s) {
    let n=s.length
    let cnt=0;
    let max=1;
    for(let i=0;i<n;i++){
        cnt=0;
        for(let j=i+1;j<n;j++){
            if(s[j-1] == s[j]){
                cnt++;
            }
            if(cnt<=1){
                max=Math.max(max,j-i+1);
            }
        }
    }
    return max;
};